src/sdl/SDL.cpp

```diff
@@ -760,36 +793,39 @@ void sdlWriteBackupStateExchange(int from, int to, int backup)
 
 void sdlWriteBattery()
 {
-    char buffer[1048];
+    char buffer[2048];
+    char *gameDir = sdlGetFilePath(filename);
+    char *gameFile = sdlGetFilename(filename);
 
     if (batteryDir)
-        sprintf(buffer, "%s/%s.sav", batteryDir, sdlGetFilename(filename));
-    else if (homeDir)
-        sprintf(buffer, "%s/%s.sav", homeDataDir, sdlGetFilename(filename));
+        sprintf(buffer, "%s%c%s.sav", batteryDir, FILE_SEP, gameFile);
+    else if (access(gameDir, W_OK) == 0)
+        sprintf(buffer, "%s%c%s.sav", gameDir, FILE_SEP, gameFile);
     else
-        sprintf(buffer, "%s.sav", filename);
+        sprintf(buffer, "%s%c%s.sav", homeDataDir, FILE_SEP, gameFile);
 
-    emulator.emuWriteBattery(buffer);
+    bool result = emulator.emuWriteBattery(buffer);
 
-    systemScreenMessage("Wrote battery");
+    if (result)
+	systemScreenMessage("Wrote battery");
```

Comment:
We probably don't want these "wrote battery" messages anymore.

See #318 

We could send them to the console instead perhaps.

Reply:
Not a necessity for this PR of course, as it's unrelated.

Reply:
It is easier to do this now rather than later.